src/server/server_family.cc

```diff
@@ -3607,18 +3608,26 @@ void ServerFamily::ReplTakeOver(CmdArgList args, const CommandContext& cmd_cntx)
     return builder->SendError("Full sync not done");
   }
 
-  std::error_code ec = replica_->TakeOver(ArgS(args, 0), save_flag);
-  if (ec)
+  Replica::TakeOverResult take_over_res = replica_->TakeOver(ArgS(args, 0), save_flag);
+  if (take_over_res == Replica::FAILED)
     return builder->SendError("Couldn't execute takeover");
 
   LOG(INFO) << "Takeover successful, promoting this instance to master.";
 
   if (IsClusterEnabled()) {
     service().cluster_family().ReconcileMasterReplicaTakeoverSlots(false);
   }
-  SetMasterFlagOnAllThreads(true);
+
+  // Start journal to allow partial sync from same source master
+  service_.proactor_pool().AwaitFiberOnAll([this](auto, auto*) { journal()->StartInThread(); });
+
   last_master_data_ = replica_->Stop();
   replica_.reset();
+  if (take_over_res == Replica::NO_PARTIAL) {
```

Comment:
why is it needed?

Reply:
To disable partial sync if all the nodes did not reach data parity before takeover completes. If we don't do that then we have data loss.